MeesGame/AI/AStar.cs

```diff
@@ -440,26 +442,20 @@ public void ThinkAboutNextAction()
         {
             Level = new AStarLevel(player.TileField);
             Level.SolveLevel();
-            if (Level.OptimalPath.Count == 0)
-                throw new PathNotFoundException();
             return Level.OptimalPath;
         }
 
         public void UpdateNextAction()
         {
-            HashSet<KeyColor> currentKeys = new HashSet<KeyColor>();
-            foreach (InventoryItem item in player.DummyPlayer.Inventory.Items)
-                currentKeys.Add(item.type.ToKeyColorType());
-            Dictionary<Point, PlayerAction> CurrentPath = path[currentKeys];
-            if (CurrentPath.ContainsKey(player.DummyPlayer.Location))
-                player.NextAIAction = CurrentPath[player.DummyPlayer.Location];
+            if(path.Count != 0)
+            {
+                HashSet<KeyColor> currentKeys = new HashSet<KeyColor>();
+                foreach (InventoryItem item in player.DummyPlayer.Inventory.Items)
```

Comment:
The property `player.DummyPlayer` creates a clone every time you call it. Try to reduce calls to it by using a local variable.